lib/pbench/server/api/resources/upload_api.py

```diff
@@ -50,70 +52,113 @@ def __init__(self, config, logger):
         )
 
     @Auth.token_auth.login_required()
-    def put(self, controller):
-
+    def put(self, filename_arg: str):
         try:
             username = Auth.token_auth.current_user().username
         except Exception:
-            self.logger.exception("Tarfile upload: Exception verifying the username")
+            self.logger.exception("Error verifying the username")
             abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
 
-        if not request.headers.get("filename"):
-            self.logger.debug(
-                "Tarfile upload: Post operation failed due to missing filename header"
+        filename = secure_filename(urllib.parse.unquote(filename_arg))
```

Comment:
Your addition of `urllib.parse.quote` for the filename is interesting; I'd thought requests would handle that automatically, but a bit of research suggests that you're right. However I believe that Flask _does_ handle decoding properly when extracting the templated parameter on the receiving end, so I think that you're double-decoding here. Unit testing to confirm this might not be a bad idea.

Reply:
Thanks for pointing this out regarding Flask behavior.

I also found out that `secure_filename()` is not going to work for us.  It munges the name so that it does not match the metadata contents.  We'd have to use `secure_filename` when making the tar ball as well.

Instead, I'll just make sure the unquoted file name does not have any path elements.  If the filename is invalid, we'll get an error trying to create it further along.